### PR TITLE
feat(typings): migrate TS typings #427

### LINKED ISSUE #427: @types/form-data: Property 'getBuffer' does not exist on type 'FormData'
The new property on `FormData`, `getBuffer` isn't properly typed in TypeScript. 

The [latest types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/form-data/index.d.ts) are the following:
```typescript
declare class FormData extends stream.Readable {
    append(key: string, value: any, options?: FormData.AppendOptions | string): void;
    getHeaders(): FormData.Headers;
    submit(params: string | FormData.SubmitOptions, callback?: (error: Error | undefined, response: http.IncomingMessage) => void): http.ClientRequest;
    getBoundary(): string;
    getLength(callback: (err: Error | undefined, length: number) => void): void;
    getLengthSync(): number;
    hasKnownLength(): boolean;
}
```

We need it to look like the following:
```typescript
declare class FormData extends stream.Readable {
    append(key: string, value: any, options?: FormData.AppendOptions | string): void;
    getHeaders(): FormData.Headers;
    getBuffer(): Buffer;
    submit(params: string | FormData.SubmitOptions, callback?: (error: Error | undefined, response: http.IncomingMessage) => void): http.ClientRequest;
    getBoundary(): string;
    getLength(callback: (err: Error | undefined, length: number) => void): void;
    getLengthSync(): number;
    hasKnownLength(): boolean;
}
```

If there isn't already a pull request open in DefinitelyTyped I can create one.

### PR DESCRIPTION
**Pull request by mapleeit**
Migrate TS typings from DefinitelyTyped/DefinitelyTyped here for the convenience of maintaining and usage.

This should fix #427

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +38,7 @@
     "node": ">= 0.12"
   },
   "dependencies": {
+    "@types/node": "^12.0.8",
```
Comment (alexindigo) [inline package.json]: @mapleeit So it's necessary "evil" to have this as prod dependency?

Comment (mapleeit) [reply package.json]: Yes, I think it is...

Otherwise people need install this by themselves when using types.

**Comment by alexindigo** [comment]
Let's get other chance to review ;)

File: package.json (hunk 1 above)
Comment (alexindigo) [reply package.json]: When they're using types, they should have it already from TS, right?

Comment (mapleeit) [reply package.json]: I'll test this later.

**Comment by mapleeit** [comment]
> Let's get other chance to review ;)

Sure, my bad.
 
--- 

Hi, @tabeth , do you have time to take a look at this. I'm not sure that is this `@types/node` dep necessary.

**Comment by alexindigo** [comment]
Thanks a lot, your help really appreciated. 🙏

**Comment by tabeth** [comment]
@alexindigo @mapleeit 

If we're going to maintain types outside of DefinitelyTyped, then yes, we do have to include `@types/node` as a dependency. Though personally I think it makes more sense to just open a pull request in DefinitelyTyped than in this repository. If we are going to continue in this way, we should to submit a PR to remove the types from DefinitelyTyped to avoid confusion.

**Comment by mapleeit** [comment]
To be honest, I'm a little confused about where is the best place to put typings files.

DefinitelyTyped is one option. But I think it may be not the best option for the convenience of usage and maintaining? (I already made a PR to DefinitelyTyped 5 days ago, still no response) And people must install `@types/form-data`.

I saw [axios](https://github.com/axios/axios) put typing file in their own repo. So I think it is an option?

---

I'll make a PR to remove the types from DefinitelyTyped if we decide to store it at here. I could also revert this commit if we decide to not, considering we still not have a new npm version.

**Comment by alexindigo** [comment]
@mapleeit @tabeth I don't any types in prod dependencies in axios – https://github.com/axios/axios/blob/master/package.json#L75-L78

**Comment by mapleeit** [comment]
@alexindigo Yes, because the typing file of `axios` is quite simple. It doesn't use `stream` and `http` definition within NodeJS. 

https://github.com/form-data/form-data/blob/master/index.d.ts#L6

**Comment by alexindigo** [comment]
This is what they have in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped):

devDep

And this is next.js server, that uses TS and relying on node modules, like `http`:
https://github.com/zeit/next.js/tree/canary/packages/next-server
– No prod types dependencies.

Also I'm seeing `@types/node` as devDep in Electron:
https://github.com/electron/electron/blob/master/package.json#L15

Let's try to put it into devDep and see how it works.

**Comment by tabeth** [comment]
@alexindigo @mapleeit 

Yeah, it would be better to put them in `devDependencies`. The types don't compile to usable code anyway, so it wouldn't actually amount to much to make them a regular dependency.

**Comment by mapleeit** [comment]
Agreed with `devDependencies`.

**Comment by mapleeit** [comment]
@tabeth @alexindigo I made a PR to move `@types/node`, please take a look when you have time : ) 

Then we may can release a version and I remove the @types/form-data from DefinitelyTyped.

**Pull request merged**